libs/designer-ui/src/lib/arrayeditor/util/util.ts

```diff
@@ -1,87 +1,109 @@
-import type { ComplexArrayItem, ComplexArrayItems, SimpleArrayItem } from '..';
+import type { ArrayItemSchema, ComplexArrayItem, ComplexArrayItems, SimpleArrayItem } from '..';
 import constants from '../../constants';
 import type { ValueSegment } from '../../editor';
+import { ValueSegmentType } from '../../editor';
+import type { CastHandler } from '../../editor/base';
 import { convertStringToSegments } from '../../editor/base/utils/editorToSegement';
 import { convertSegmentsToString } from '../../editor/base/utils/parsesegments';
-import { getIntl } from '@microsoft/intl-logic-apps';
 import { guid } from '@microsoft/utils-logic-apps';
 
 export interface ItemSchemaItemProps {
+  key: string;
   title: string;
   type: string;
   isRequired: boolean;
   description: string;
-  key: string;
+  format?: string;
+  items?: ItemSchemaItemProps[];
 }
 
-export const getOneDimensionalSchema = (itemSchema: any, isRequired?: any): ItemSchemaItemProps[] => {
+export const getOneDimensionalSchema = (itemSchema: ArrayItemSchema, isRequired?: any): ItemSchemaItemProps[] => {
   const flattenedSchema: ItemSchemaItemProps[] = [];
   if (!itemSchema) {
     return flattenedSchema;
   }
   if (itemSchema.type === constants.SWAGGER.TYPE.OBJECT && itemSchema.properties) {
     const required = itemSchema.required ?? [];
-    Object.keys(itemSchema.properties).forEach((key) => {
-      const value = itemSchema.properties[key];
-      if (value) {
+    Object.entries(itemSchema.properties).forEach(([key, value]) => {
+      if (value && key !== 'key') {
         getOneDimensionalSchema(value, required.includes(key)).forEach((item) => {
           const currItem = item;
-          if (!item.key) {
-            currItem.key = key;
-          }
           flattenedSchema.push(currItem);
         });
       }
     });
   } else {
+    const isArray = itemSchema.type === constants.SWAGGER.TYPE.ARRAY && itemSchema.items?.properties;
     flattenedSchema.push({
-      title: itemSchema.title,
-      type: itemSchema.type,
-      isRequired: isRequired ?? false,
+      key: itemSchema.key,
+      title: itemSchema.title ?? itemSchema.key.split('.').slice(-1)[0],
+      type: isArray ? constants.SWAGGER.TYPE.ARRAY : itemSchema.type,
```

Comment:
Why isn't itemSchema.type set to array for nested array?

Reply:
it should be, I think I had set this earlier just to enforce that arrays should be arrays hahaha. Will just set it to be       type: itemSchema.type,